app/javascript/src/ChangePassword/components/ChangePassword.jsx

```diff
@@ -0,0 +1,106 @@
+// @flow
+
+import React, { useState } from 'react'
+import { createReactWrapper } from 'utilities/createReactWrapper'
+import { LoginPage, Form, ActionGroup, Button } from '@patternfly/react-core'
+import { HiddenInputs, FlashMessages } from 'LoginPage'
+import { PasswordInput } from 'ChangePassword'
+import type { FlashMessage } from 'Types'
+import { validateForm } from 'LoginPage/utils/formValidation'
+
+import 'LoginPage/assets/styles/loginPage.scss'
+import brandImg from 'LoginPage/assets/images/3scale_Logo_Reverse.png'
+import PF4DownstreamBG from 'LoginPage/assets/images/PF4DownstreamBG.svg'
+
+type Props = {
+  lostPasswordToken: ?string,
+  url: string,
+  errors: (?FlashMessage)[]
+}
+
+const validationConstraints = {
+  'user[password]': {
+    presence: true,
+    length: { minimum: 1 }
+  },
+  'user[password_confirmation]': {
+    presence: true,
+    length: { minimum: 1 },
+    equality: 'user[password]'
+  }
+}
+
+const ChangePassword = ({ lostPasswordToken, url, errors }: Props) => {
+  const [isFormDisabled, setIsFormDisabled] = useState(true)
+  const [passwordsMatch, setPasswordsMatch] = useState(true)
+
+  const onChange = (event) => {
+    const errors = validateForm(event.currentTarget, validationConstraints)
+
+    if (errors) {
```

Comment:
This whole block is a bit weird. We are validating whether there are errors or not, but then we are just asserting an specific one, that only this one will set the condition of disabling the form. Isn't there any other way that it could be more generic and avoiding this specificity? Also, it's calling the same fn again in the else case, but opposite values, that could be definitely DRY up.